### ISSUE TITLE
Why is my model not learning for A2C but working with PPO?

### ISSUE THREAD

**Issue by Harishu1998** [DLR-RM/stable-baselines3#1590]
### ❓ Question

I have a very simple fermentation custom environment which I trained on A2C and PPO.  I do not have a very good understanding of these 2 algorithms but I read that PPO is a variant of A2C. So, Ideally if PPO is working then A2C should also work right? 

My environment:

objective is reach the highest fermentation product concentration possible by changing its temperature. 
action space : Discrete(3) where
```
0: do not change temperature, 
1: increase temperature by 0.5 degrees and 
2: reduce temperature by 0.5 degrees.
``` 
observation space : Box(low= 0 , high = 10) this is the production concentration that we are observing at each step() 
reward: slope * 100 (if the slope is small then reward will be small but if slope is steep then reward will be high  and if slope is negative the program stops and a negative reward is given)

This environment terminates if there is a decrease in product concentration (-ve slope) or when the experiement time is over (1200 steps)

when I trained it on A2C for ~5M timesteps I see that the ep_mean_reward is reducing and becomes a constant after a while

![image](https://github.com/DLR-RM/stable-baselines3/assets/113226646/9c110d29-3e8a-4e15-ac51-6ab6f48dc293)

But, the same environment when trained on PPO shows good results ie. ep_mean_reward was increasing over time 

![image](https://github.com/DLR-RM/stable-baselines3/assets/113226646/d11093de-75d8-4149-821c-c9c066d0ca15)

But, eventhough PPO was learning over time I observed from the graphs that 
1. Entropy_loss was increasing over time 
2. learning rate was constant 
3. train/loss was increasing 
4. policy_gradient_loss was like a mountain which was decreased towards the end
5. value_loss was increasing gradually

I believe the loss should always reduce if the mean_episode_reward is increasing over time. 

```
class GridWorldEnv(gymnasium.Env):

    def __init__(self):
        self.t0 = 0.0  # initial time 
        self.tf = 120  # final time minutes
        self.dt = 0.1  # time step
        self.observation_space  = spaces.Box(low=np.array([0]), high = np.array([20]), dtype=np.float64)
        self.action_space = spaces.Box(low = np.array([0]), high=np.array([2]), dtype=np.int16)
        self.X0 = 0.3e5  # cell/ml
        self.S0 = 0.2 # g/L
        self.E0 = 0 # U/L 
        self.i = 0

        # Model Parameters
        self.Ks = 0.1 #g/L substrate saturation coefficient
        self.C = 0.000001 # ug/cell - glucose consumption per new cell created (growth coefficient)
        self.MuX = 0.1 # 1/hr
        self.MuE = .000001 # U/(cell*hr)
        self.MuD = 0 # 1/hr
        self.mu_opt = 1.8
        self.T_opt = 37
        self.A_opt = 250
        self.r_t = 3.12
        self.r_a = 4.5
        self.temperature = np.random.randint(low=32, high=35)

        # Substrate Values
        self.S = np.zeros(int(self.tf/self.dt)+1)
        # Initial Substrate
        self.S[0] = self.S0

        # Cell Values
        self.X = np.zeros(int(self.tf/self.dt)+1)
        # initial Cell concentration value
        self.X[0] = self.X0

        # Enzyme concentration
        self.E0 = 0
        self.E = np.zeros(int(self.tf/self.dt)+1)
        self.E[0] = self.E0

        # Time steps
        self.t = np.arange(self.t0, self.tf+self.dt, self.dt)

        # Change in Enzyme 
        self.E_C = np.zeros(int(self.tf/self.dt))

       
        

    def reset(self, seed=None, options=None):
        # We need the following line to seed self.np_random
        super().reset(seed=seed)
        self.t0 = 0.0  # initial time
        self.tf = 120  # final time
        self.dt = 0.1  # time step
        self.i = 0
        self.X0 = 0.3e5  # cell/ml
        self.S0 = 0.2 # g/L
        self.E0 = 0 # U/L 
        # Process conditions

        # Model Parameters
        self.Ks = 0.1 #g/L substrate saturation coefficient
        self.C = 0.000001 # ug/cell - glucose consumption per new cell created (growth coefficient)
        self.MuX = 0.1 # 1/hr
        self.MuE = .000001 # U/(cell*hr)
        self.MuD = 0 # 1/hr
        self.mu_opt = 1.8
        self.T_opt = 37
        self.A_opt = 250
        self.r_t = 3.12
        self.r_a = 4.5
        self.temperature = np.random.randint(low=32, high=35)

        # Substrate Values
        self.S = np.zeros(int(self.tf/self.dt)+1)
        # Initial Substrate
        self.S[0] = self.S0

        # Cell Values
        self.X = np.zeros(int(self.tf/self.dt)+1)
        # initial Cell concentration value
        self.X[0] = self.X0

        # Enzyme concentration
        self.E0 = 0
        self.E = np.zeros(int(self.tf/self.dt)+1)
        self.E[0] = self.E0

        # Time steps
        self.t = np.arange(self.t0, self.tf+self.dt, self.dt)

        # Change in Enzyme 
        self.E_C = np.zeros(int(self.tf/self.dt))

        # Reset makes the enzyme concetration 0
        observation = np.array([self.E0],dtype=float)
        info = {}


        with open("plot.txt",'w') as f:
            f.truncate(0)
            f.close()

        return observation, info


    def step(self, action):
        # Termincation condition
        if self.i == 1199:
            terminate = True
            return np.array([self.E[self.i]]), 0, terminate, False, {} 
        else:
            terminate = False
            action = math.ceil(action[0])
            initial_cordinates = [self.i, self.E[self.i]]  
            if action == 2:
                self.temperature -= 0.1
            else:
                self.temperature += action/10

            MuX =  self.mu_opt*(math.exp(-((self.temperature - self.T_opt)**2)/self.r_t**2))

            dXdt = (MuX *self.S[self.i]) / (self.Ks + self.S[self.i]) * self.X[self.i]
            
            dSdt = ( -self.C * MuX*self.S[self.i]/(self.Ks + self.S[self.i]) * self.X[self.i] )

            delX = dXdt * self.dt
            delS = dSdt * self.dt

            self.X[self.i+1] = self.X[self.i] + delX
            self.S[self.i+1] = self.S[self.i] + delS

            delE = self.MuE * dXdt * self.dt

            step = self.i
            nts = 1

            while step > 0 and nts < 500:
                delE = delE + (self.X[self.i+1] - self.X[self.i]) * (self.MuE)*(1/nts)
                nts += 1
                step -= 1
            
            self.E_C[self.i] = delE
            self.E[self.i+1] = self.E[self.i] +  delE

            if self.i+1 > 51:
                self.E[self.i+1] = self.E[self.i+1] - self.E_C[self.i-50]

            final_cordinates = [self.i+1, self.E[self.i+1]]
            
            slope = (final_cordinates[1] - initial_cordinates[1]) / (final_cordinates[0] - initial_cordinates[0])

            if slope < 0:
                reward = 10 * slope
                terminate = True
            else:
                slope = 100 * slope
                reward = 10 + slope

            with open('plot.txt','a') as plotting_file:
                plotting_file.write(f"{self.i},{self.X[self.i]},{self.S[self.i]},{self.E[self.i]}\n")
                plotting_file.close()
            self.i += 1
            return np.array([self.E[self.i+1]]), reward, terminate, False, {}

    def render(self):
        pass

    
    def close(self):
        pass

```

I am not able to find an explanation to this behavior. Any help is highly appreciated. 

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Hello,
actually, it is the other way around, a2c can be seen as a spécial case of ppo: https://arxiv.org/abs/2205.09123

And it's a classic result that ppo tends to be easier to tune on tasks (see open rl benchmark for instance).
And you can take a look at the links in our doc to learn more about what is special about ppo ;)
